flow/connectors/clickhouse/normalize.go

```diff
@@ -526,38 +527,39 @@ func (c *ClickHouseConnector) NormalizeRecords(
 				if insertIntoSelectQuery.Part == numParts-1 {
 					c.logger.Info("[clickhouse] set last normalized batch id for table",
 						slog.String("table", insertIntoSelectQuery.TableName),
-						slog.Int64("syncBatchID", endBatchID),
-						slog.Int64("lastNormalizedBatchID", normBatchID),
+						slog.Int64("endBatchId", insertIntoSelectQuery.endBatchId),
+						slog.Int64("startBatchId", insertIntoSelectQuery.startBatchId),
 						slog.Int("parallelWorker", i))
-					err := c.SetLastNormalizedBatchIDForTable(ctx, req.FlowJobName, insertIntoSelectQuery.TableName, endBatchID)
+					err := c.SetLastNormalizedBatchIDForTable(ctx, req.FlowJobName, insertIntoSelectQuery.TableName, insertIntoSelectQuery.endBatchId)
```

Comment:
Seems like this could run before a previous one runs in parallel? Great that we're reviewing the code like this

Reply:
The queries are parallelized by table, so processing out-of-order across tables is acceptable